### ISSUE TITLE
gadgets/deadlock: Use gadget_process structure

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [inspektor-gadget/inspektor-gadget#3740]
https://github.com/inspektor-gadget/inspektor-gadget/commit/d4f534d4991fab90b8835dd5b77eb52a80e44207 introduced and used a gadget_process structure on different gadgets, this was changed back on the deadlock gadget by #3532. The goal of this issue is to use such structure again as mentioned in https://github.com/inspektor-gadget/inspektor-gadget/pull/3532#discussion_r1842330676

**Comment by Snehil-Shah** [comment]
@mauriciovasquezbernal The reason I changed that is because the map structure is as follows:

```
key = { pid, mutex1, mutex2 }
value = { tid, mntns_id, comm, stack_ids }
```

Using `gadget_process` struct will result in unnecessary duplication of the PID field. Moreover, if we wish to use the `gadget_populate` helper method to auto-fill the gadget_process struct, we would also have to do that twice for both the key and value, which is just unnecessary perf overhead. I think for this gadget, it makes sense to stay conservative on this.

**Comment by mauriciovasquezbernal** [comment]
I think the key can be the same as before, perhaps using the `__` prefix on the pid field to avoid field duplication (as done in the [top file](https://github.com/inspektor-gadget/inspektor-gadget/blob/e86eefff1f80e50229609a5e3ff0639f20674b2f/gadgets/top_file/program.bpf.c#L31-L35) gadget), then the value could contains the gadget_process struct. I don't think the duplication of the pid field is a big deal on this case, IMO it's better to use common helpers.

**Comment by Snehil-Shah** [comment]
But wouldn't we have to get the `PID` twice in this case? Once using `gadget_populate` on the value and once to set the `PID` in the key?

**Comment by Snehil-Shah** [comment]
Also, I am sending the map entries to wasm space, and I'm not sure how to decode the pid, tid, comm etc from the `gadget_process` struct using `GetField`..?

**Comment by mauriciovasquezbernal** [comment]
> But wouldn't we have to get the `PID` twice in this case? Once using `gadget_populate` on the value and once to set the `PID` in the key?

That's right. Two calls to `bpf_get_current_pid_tgid` are needed, one for the key and another one for the value in `gadget_populate`. This helper should be very fast and I think the overhead of calling it twice is negligible. We can check other folks opinions on this one, cc @alban 

> Also, I am sending the map entries to wasm space, and I'm not sure how to decode the pid, tid, comm etc from the `gadget_process` struct using `GetField`..?

`GetField("proc.pid")`, etc. should work.

**Comment by farhanashrafdev** [comment]
@mauriciovasquezbernal I will like to take this up, can you assign me
